### PR TITLE
Update Dev docs

### PR DESCRIPTION
**Pull request by ekumachidi**
### Trello card
https://trello.com/c/S08VmyK0/5913-review-and-update-developer-documentation-for-get-school-experience

### Context
Update the dev docs to be more current with the changes in the application.

### Changes proposed in this pull request
* Update README.md with updated Ruby and Bundler versions, NodeJS version, and other dependencies
* Update DfE Sign-in integration documentation
* Add documentation for inviting users to school's DFE Sign in via Manage School Dashboard

### CODE REVIEW

File: docs/documents/mse-user-invite.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+It also meant that we would be required to build a support console in the near future so that
+the GIT support team and the Schools could manage user as well as candidate requests.
+
+We found a way to achieve the user management with the existing version of DSI by integrating
+the `public api` into the Manage School Experience service.
+
+Users will be able to invite new users via School Dashboard in Manage School Experience
+
+The new users will be assigned an organisation they have been invite to manage and will be
```
Comment (martyn-w) [inline docs/documents/mse-user-invite.md:31]: typo: "... have been INVITED to manage..."

File: docs/documents/mse-user-invite.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+assigned an ‘End-user’ role by default.
+
+## Caveat:
+
+- DSI only assigns an ‘end-user’ role via public API - security protocol
+- DSI also does not assign service by default via public API - security protocol
+
+## Final solution
```
Comment (martyn-w) [inline docs/documents/mse-user-invite.md:39]: "Current solution" reads better

File: docs/documents/mse-user-invite.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Invite users to school's DFE Sign in via Manage School Dashboard
+
+## Identify only vs Identity and Access Management
+
+Current DSI implementation on Manage School Experience is Identity and Access Management
```
Comment (martyn-w) [inline docs/documents/mse-user-invite.md:5]: maybe clarify that DSI = "DfE Sign In" as it is not mentioned elsewhere

**Review by martyn-w — APPROVED** [review]
👍  LGTM

**Review by MylesJarvis — APPROVED** [review]
This looks fine to me

**Pull request merged**